zerver/tests/test_settings.py

```diff
@@ -268,24 +269,40 @@ def do_test_change_user_display_setting(self, setting_name: str) -> None:
         # Error if a setting in UserProfile.property_types does not have test values
         if test_value is None:
             raise AssertionError('No test created for %s' % (setting_name))
-        invalid_value = 'invalid_' + setting_name
 
-        data = {setting_name: ujson.dumps(test_value)}
+        if setting_name == 'demote_inactive_streams':
+            invalid_int_value = 4
+            data = {setting_name: test_value}
+        else:
+            invalid_value = 'invalid_' + setting_name
+            data = {setting_name: ujson.dumps(test_value)}
+
         result = self.client_patch("/json/settings/display", data)
         self.assert_json_success(result)
         user_profile = self.example_user('hamlet')
         self.assertEqual(getattr(user_profile, setting_name), test_value)
 
         # Test to make sure invalid settings are not accepted
         # and saved in the db.
-        data = {setting_name: ujson.dumps(invalid_value)}
+        if setting_name == 'demote_inactive_streams':
+            data = {setting_name: invalid_int_value}
+        else:
+            data = {setting_name: ujson.dumps(invalid_value)}
```

Comment:
I think if you just set `invalid_value` differently, you wouldn't need this conditional.

Reply:
If i use the same variable for both str and int the backend lint fails giving the error `Incompatible types in assignment`.

Reply:
You can fix this with a type annotation `# type: Union[int, str]` on the line the variable is declared